python_modules/libraries/dagster-airlift/dagster_airlift/core/monitoring_job/utils.py

```diff
@@ -61,3 +72,51 @@ def get_dagster_run_for_airflow_repr(
         ),
         None,
     )
+
+
+_T = TypeVar("_T")
+
+
+def _assert_param_value(value: _T, expected_values: Iterable[_T]) -> _T:
+    if value not in expected_values:
+        raise Exception(
+            f"Invalid value when translating metadata from logs. Expected one of"
+            f" `{expected_values}`, got `{value}`."
+        )
+    return value
+
+
+def extract_metadata_from_logs(context: OpExecutionContext, logs: str) -> dict[str, MetadataValue]:
+    metadata = {}
+    import re
+
+    matches = re.findall(r"DAGSTER_START(.*?)DAGSTER_END", logs, re.DOTALL)
+    for match in matches:
+        raw_external_metadata_map = json.loads(match)
```

Comment:
I think we need to swallow errors here -- otherwise if the user emits invalid metadata a single time then their sync job will be stuck right?